# ***`TEMEL OLASILIK`***
**Problem :** Bir spor salonuna günde 1000 kişi spor yapmak için geliyor. Gelen kişilerin %30'ı kilo vermek istiyor. Sabah saatlerinde (08:00-12:00 saatleri arasında) kulübe 50 kişi geliyor.
Sabah saatlerinde kulübe gelen kişilerden kilo vermek isteyen bir kişiyle karşılaşma olasılığımız nedir?

p (Toplam Kişi) = 1000

s (Sabah saatlerinde gelen kişi sayısı) = 50

f (Kilo Vermek İsteyen Kişi Sayısı) = 1000 * 30 / 100 = 300

Her kişi ile karşılaşma eşit şanslı olay olursa:

1 değerinden istenmeyen durumların gerçekleşme olasılığını çıkarırsak sonuca ulaşabiliriz.

P (kilo vermek isteyen kişi sayısı) = 1 - C (p-f,s) / C (p,s)

In [1]:
print("Kilo vermek isteyen kişi sayısı:",1000 * 30 / 100)

Kilo vermek isteyen kişi sayısı: 300.0


In [5]:
def faktoriyel(sayi):
  sonuc=1
  for i in range(1,sayi+1):
    sonuc=sonuc*i
  return sonuc
  
def kombinasyon(n,r):
  return faktoriyel(n)/(faktoriyel(n-r)*faktoriyel(r))

In [6]:
p=3000
s=50
f=1200
k=p-f

P=1-kombinasyon(k,s)/kombinasyon(p,s)
print(P)
     
     

0.9999999999938683


**SONUÇ:**Sabah saatlerinde spor kulübüne gelen kişilerden kilo vermek isteyen bir kişiyle karşılaşma olasılığımız 0.9999999999938683 olarak bulunur.